### PR TITLE
SQLite CVE-2023-32697  fix

### PR DESCRIPTION
**Pull request by kelvinqian00**
Fix CVE-2023-32697, which is imperative as it is a remote code execution vulnerability.

### CODE REVIEW

**Comment by kelvinqian00** [comment]
Not sure how the above 4 commits got in here, but since they were merged cleanly I don't think they should be a problem.

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@ pom.xml
 pom.xml.asc
 *.jar
 *.class
+*.pem
```
[the remark this answers is not in the document]
Comment (cliffcaseyyet) [reply .gitignore:6]: mmm, i dunno. It's fine but generally .gitignore is not for "i happen to put this random file here and don't want it" its for "in the course of general development this gets created by some automatic process or IDE and i dont want it in the repo"

[the remark this answers is not in the document]
Comment (kelvinqian00) [reply .gitignore:6]: Sure, but in this case we _really_ don't want PEM files showing up in the public repo so this is also a "better safe than sorry" thing.

[the remark this answers is not in the document]
Comment (kelvinqian00) [reply .gitignore:6]: For what it's worth, I also had a random `.txt` file but I _didn't_ add `*.txt` to the gitignore, for much the same reason you said.

[the remark this answers is not in the document]
Comment (cliffcaseyyet) [reply .gitignore:6]: Yeah, just because of the specific extension itself I approved already.

**Pull request merged**